util/run-gnu-test.sh

```diff
@@ -33,7 +33,18 @@ if test $# -ge 1; then
     # if set, run only the tests passed
     SPECIFIC_TESTS=""
     for t in "$@"; do
-        SPECIFIC_TESTS="$SPECIFIC_TESTS $t"
+
+        # Construct the full path
+        full_path="$path_GNU/$t"
+
+
+        # Check if the file exists with .sh, .pl extension or without any extension in the $path_GNU directory
+        if [ -f "$full_path" ] || [ -f "$full_path.sh" ] || [ -f "$full_path.pl" ]; then
+            SPECIFIC_TESTS="$SPECIFIC_TESTS $t"
+        else
+            echo "Warning: Test file $full_path, $full_path.sh, or $full_path.pl does not exist!"
+            sleep 1s
```

Comment:
A not existing file is an error imo and so I would exit the script right away, otherwise the error message is "hidden" at the beginning and easy to overlook with all the other output. And if you specify a single file, the test output is green, signaling success.
```suggestion
            echo "Error: Test file $full_path, $full_path.sh, or $full_path.pl does not exist!"
            exit 1
```

Reply:
This is why i put a sleep but you are probably correct